yearclass/src/main/java/com/facebook/device/yearclass/YearClass.java

```diff
@@ -132,6 +138,19 @@ private static int getNumCoresYear() {
   }
 
   /**
+   * Evaluates the processor's type
+   * @return a String which contains the processor's name
+   */
+  public static String getCoreDescription() {
+    int cores = DeviceInfo.getNumberOfCPUCores();
+    if (cores < 1) return UNKNOWN;
```

Comment:
Shouldn't this be in reverse order? Starting with cores == 4, working down to the else case where it would be unknown? This way it's future proof because more cores will not be incorrectly reported as quad core but as unknown instead.